src/rebar_app_discover.erl

```diff
@@ -16,16 +16,33 @@ do(State, LibDirs) ->
     Dirs = [filename:join(BaseDir, LibDir) || LibDir <- LibDirs],
     Apps = find_apps(Dirs, all),
     ProjectDeps = rebar_state:deps_names(State),
+
     lists:foldl(fun(AppInfo, StateAcc) ->
+                        StateAcc1 = merge_deps(AppInfo, StateAcc),
```

Comment:
This call likely means that we need to ensure the app-level deps are traversed in the stable-sorted order of apps.

If `A` and `B` both depend on `C`, the `C` from `A` must be taken. Is this taken care of here? It doesn't seem to explicitly take place.